extensions/cache/deployment/src/main/java/io/quarkus/cache/deployment/CacheProcessor.java

```diff
@@ -74,9 +75,9 @@ ValidationErrorBuildItem validateBeanDeployment(ValidationPhaseBuildItem validat
     }
 
     @BuildStep
-    @Record(STATIC_INIT)
+    @Record(RUNTIME_INIT)
```

Comment:
We need to manually propagate the context using the Quarkus `ManagedExecutor` to fix #8166 and this executor is initialized in generated bytecode executed at `ExecutionTime.RUNTIME_INIT`. That's why the caches intialization was moved from `STATIC_INIT` to `RUNTIME_INIT` here.